[DOC] Create Dashboards quickstart (#2409)
opensearch-project/documentation-website: _dashboards/get-started/quickstart-dashboards.md

---
layout: default
title: Quickstart for OpenSearch Dashboards
nav_order: 20
has_children: false
---

# Quickstart for OpenSearch Dashboards

This quickstart covers the core concepts for getting started with OpenSearch Dashboards. You'll learn how to:

> - Add sample data

Reviewer: Do we also need to include a section on guiding users to add their own data? so not just sample data

Author: Based on our chat, we'll  leave the quickstart with sample data. A separate "Adding data" section about adding your own data will be created.

- Explore and inspect data with **Discover**
- Visualize data with **Dashboard**

Before you get started, make sure you've installed OpenSearch and OpenSearch Dashboards. For information on installation and configuration, see [Install and configure OpenSearch]({{site.url}}{{site.baseurl}}/install-and-configure/install-opensearch/index/) and [Install and configure OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/install-and-configure/install-dashboards/index/).
{: .note}

# Add sample data

Sample data sets come with visualizations, dashboards, and more to help you explore Dashboards before you add your own data. To add sample data, perform the following steps:

1. Verify access to OpenSearch Dashboards by connecting to [http://localhost:5601](http://localhost:5601) from a browser. The default username and password are `admin`.
2. On the OpenSearch Dashboards **Home** page, select **Add sample data**.
3. Select **Add data** to add the data sets.

> ![Sample data sets]({{site.url}}{{site.baseurl}}/images/add-sample-data.png)

Reviewer 2: ```suggestion
<img src="({{site.url}}{{site.baseurl}}/images/add-sample-data.png" alt="Sample data sets">{: .img-fluid}
```

Reviewer 2: This suggestion also applies to any other image links with "!" at the beginning in the document.

# Explore and inspect data

In [**Discover**](discover/index.md), you can: 

- Select data to explore, set a time range for that data, search it using [Dashboards Query Language (DQL)]({{site.url}}{{site.baseurl}}/dashboards/dql/), and filter the results.
- Explore the data's details, view individual documents, and create tables summarizing the data's contents.
- Visualize your findings.

## Try it: Get familiar with Discover

1. On the OpenSearch Dashboards **Home** page, select **Discover.**
1. Change the [time filter](time-filter.md) to **Last 7 days**.
![Time filter interface]({{site.url}}{{site.baseurl}}/images/last-7--days.png)
1. To view US-bound flights delayed 60 minutes or longer in a specific region, use the DQL search field:
`FlightDelay:true AND DestCountry: US AND FlightDelayMin >= 60`
1. To filter data by delay type, select the field under **Available field**.

> ![Filter data interface]({{site.url}}{{site.baseurl}}/images/filter-data.png)

Author: Replaced image as suggested.

# Visualize the data

Raw data can be hard to comprehend and use, and data visualizations help you prepare and present data in a visual form. In **Dashboard** you can:

- Display data in a single view.
- Build dynamic dashboards.
- Create and share reports.
- Embed analytics to differentiate your applications.

## Try it: Get familiar with Dashboard

1. On the OpenSearch Dashboards **Home** page, select **Dashboard**.
1. Select **[Flights] Global Flight Data** in the **Dashboards** window.
![Data visualization dashboard]({{site.url}}{{site.baseurl}}/images/dashboard-flight-quickstart.png)

> 

Author: Added a step to explore further by adding a panel

## Try it: Create a visualization panel

Continuing with the preceding dashboard, you'll create a bar chart comparing the number of cancelled flights and delayed flights to delay type, and then add the panel to the dashboard.

> 1. Change the default [time range](time-filter.md) from **24 hours** to **Last 7 days**. 

Reviewer 3: ```suggestion
1. Change the default [time range]({{site.url}}{{site.baseurl}}/dashboards/get-started/time-filter) from **24 hours** to **Last 7 days**. 
```

1. In the toolbar, select **Edit**, then **Create new**.
1. Select **VisBuilder** in the **New Visualizations** window.
1. In the **Data Source** dropdown list, select `opensearch_dashboards_sample_data_flights`.
1. Drag the fields **Cancelled** and **FlightDelay** to the y-axis column.
1. Drag the field **FlightDelayType** to the x-axis column.
1. Select **Save** and name the visualization in the **Title** field.

> 1. Select *Save and return**. The bar chart is added as the last panel to the dashboard. 

Reviewer 3: ```suggestion
1. Select **Save and return**. The bar chart is added as the last panel to the dashboard. 
```

![Bar chart visualization panel]({{site.url}}{{site.baseurl}}/images/viz-panel-quickstart.png)

# Interact with the data

Interactive dashboards allow you analyze data more deeply and filter it several ways. In Dashboards, you can interact directly with data on a dashboard by using dashboard-level filters. For example, continuing with the preceding dashboard, you can filter to show delays and cancellations for a specific airline.

## Try it: Interact with the flight data

> 1. On the **[Flights] Airline Carrier** panel, select **OpenSearch-Air**. The dashboard updates automatically .

Reviewer 3: ```suggestion
1. On the **[Flights] Airline Carrier** panel, select **OpenSearch-Air**. The dashboard updates automatically.
```

1. Select **Save** to save the customized dashboard.

Alternatively, you can apply filters using the dashboard toolbar:

1. In the dashboard toolbar, select **Add filter**.
1. From the **Field**, **Operator**, and **Value** dropdown lists, select **Carrier**, **is**, and **OpenSearch-Air**, respectively.
![Edit field interface]({{site.url}}{{site.baseurl}}/images/edit-filter.png)
1. Select **Save**. The dashboard updates automatically.

The result is the following dashboard: 

![Dashboard view after applying Carrier filter]({{site.url}}{{site.baseurl}}/images/interact-filter-dashboard.png)

# Next steps

> **Visualize data**. Learn more about data visualizations in OpenSearch Dashboards. Go to **[Visualize]**({{site.url}}{{site.baseurl}}/dashboards/visualize/viz-index/).

Reviewer 3: ```suggestion
**Visualize data**. Learn more about data visualizations in OpenSearch Dashboards. Go to [**Visualize**]({{site.url}}{{site.baseurl}}/dashboards/visualize/viz-index/).
```

> **Create dashboards.** Learn more about visualizing data in OpenSearch Dashboards. Go to **[Dashboard]**(add link-content in development).

Reviewer 3: ```suggestion
**Create dashboards.** Learn more about visualizing data in OpenSearch Dashboards. Go to [**Dashboard**](add link-content in development).
```

> **Explore data**. Learn more about exploring data in OpenSearch Dashboards. Go to **[Discover]**({{site.url}}{{site.baseurl}}/dashboards/discover/). 

Reviewer 3: ```suggestion
**Explore data**. Learn more about exploring data in OpenSearch Dashboards. Go to [**Discover**]({{site.url}}{{site.baseurl}}/dashboards/discover/). 
```

This quickstart guide covers the core concepts that you need to understand to get started with OpenSearch Dashboards. You'll learn how to:

- Add sample data.
- Explore and inspect data with **Discover**.
- Visualize data with **Dashboard**.
- Add sample data.
- Explore and inspect data with **Discover**.
- Visualize data with **Dashboard**.

Before you get started, make sure you've installed OpenSearch and OpenSearch Dashboards. For information on installation and configuration, see [Install and configure OpenSearch]({{site.url}}{{site.baseurl}}/install-and-configure/install-opensearch/index/) and [Install and configure OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/install-and-configure/install-dashboards/index/).
{: .note}

# Adding sample data

Sample datasets come with visualizations, dashboards, and other tools to help you explore Dashboards before you add your own data. To add sample data, perform the following steps:

1. Verify access to OpenSearch Dashboards by connecting to [http://localhost:5601](http://localhost:5601) from a browser. The default username and password are `admin`.
1. On the OpenSearch Dashboards **Home** page, choose **Add sample data**.

> 1. Choose **Add data** to add the data sets, as shown in the following image.

Reviewer 4: ```suggestion
1. Choose **Add data** to add the datasets, as shown in the following image.
```

Author: Change made

    <img src="{{site.url}}{{site.baseurl}}/images/add-sample-data.png" alt="Sample data sets" width="800">

# Exploring and inspecting data

In [**Discover**](discover/index.md), you can: 

- Choose data to explore, set a time range for that data, search it using [Dashboards Query Language (DQL)]({{site.url}}{{site.baseurl}}/dashboards/dql/), and filter the results.
- Explore the data, view individual documents, and create tables summarizing the data's contents.
- Visualize your findings.

## Try it: Getting familiar with Discover

> 1. On the OpenSearch Dashboards **Home** page, choose **Discover.**

Reviewer 4: Does "Discover" include a period in the UI? If not, the period should not be in bold.

Author: Change made

1. Change the [time filter](time-filter.md) to **Last 7 days**, as shown in the following image.

    <img src="{{site.url}}{{site.baseurl}}/images/last-7--days.png" alt="Time filter interface" width="350"/>

2. Search using the DQL query `FlightDelay:true AND DestCountry: US AND FlightDelayMin >= 60` and then choose **Update**. You should see results for US-bound flights delayed by 60 minutes or more, as shown in the following image.
   
    <img src="{{site.url}}{{site.baseurl}}/images/dashboards/dql-search-field.png" alt="DQL search field example" width="1000"/>

> 3. To filter data, choose **Add filter** and then select an **Available field**. For example, select `FlightDelayType`, **is**, and **Weather delay** from the **Field, Operator, and Value** dropdown lists, as shown in the following image.

Reviewer 4: If Field, Operator, and Value are separate dropdown lists, the commas separating the terms should not be in bold.

Author: Change made

    <img src="{{site.url}}{{site.baseurl}}/images/dashboards/filter-data-discover.png" alt="Filter data by FlightDelayType field" width="350"/>

# Visualizing data

Raw data can be difficult to comprehend and use. Data visualizations help you prepare and present data in a visual form. In **Dashboard** you can:

- Display data in a single view.
- Build dynamic dashboards.
- Create and share reports.
- Embed analytics to differentiate your applications.

## Try it: Getting familiar with Dashboard

1. On the OpenSearch Dashboards **Home** page, choose **Dashboard**.
1. Choose **[Flights] Global Flight Data** in the **Dashboards** window, as shown in the following image.

    <img src="{{site.url}}{{site.baseurl}}/images/dashboards/dashboard-flight-quickstart.png" alt="Data visualization dashboard" width="800"/>

1. To add panels to the dashboard, choose **Edit** and then **Add** from the toolbar.
1. In the **Add panels** window, choose the existing panel **[Flights] Delay Buckets**. You'll see a pop-up window on the lower right confirming that you've added the panel.

> 1. Select `x` to close the Add panels window.

Reviewer 4: Let's put "Add panels" in bold here.

Author: Change made

1. View the added panel **[Flights] Delay Buckets**, which is added as the last panel on the dashboard, as shown in the following image.

    <img src="{{site.url}}{{site.baseurl}}/images/dashboards/add-panel.png" alt="Add panel to dashboard" width="600"/>

## Try it: Creating a visualization panel

> Continuing with the preceding dashboard, you'll create a bar chart comparing the number of canceled flights and delayed flights to delay type and then add the panel to the dashboard.

Reviewer 4: Replace the terminating period with a colon.

Author: Change made

Interactive dashboards allow you analyze data in more depth and filter it in several ways. In Dashboards, you can interact directly with data on a dashboard by using dashboard-level filters. For example, continuing with the preceding dashboard, you can filter to show delays and cancellations for a specific airline.

## Try it: Interacting with the sample flight data

1. On the **[Flights] Airline Carrier** panel, choose **OpenSearch-Air**. The dashboard updates automatically.
1. Choose **Save** to save the customized dashboard.

Alternatively, you can apply filters using the dashboard toolbar:

1. In the dashboard toolbar, choose **Add filter**.
1. From the **Field**, **Operator**, and **Value** dropdown lists, choose **Carrier**, **is**, and **OpenSearch-Air**, respectively, as shown in the following image.

    <img src="{{site.url}}{{site.baseurl}}/images/edit-filter.png" alt="Edit field interface" width="400"/>

1. Choose **Save**. The dashboard updates automatically, and the result is the dashboard shown in the following image.

  <img src="{{site.url}}{{site.baseurl}}/images/interact-filter-dashboard.png" alt="Dashboard view after applying Carrier filter" width="800"/>

> # Taking the next steps

Reviewer 4: ```suggestion
# Next steps
```

Author: Change made

- **Visualize data**. To learn more about data visualizations in OpenSearch Dashboards, see [**Visualize**]({{site.url}}{{site.baseurl}}/dashboards/visualize/viz-index/).

> - **Create dashboards.** To learn more about creating dashboards in OpenSearch Dashboards, see **Dashboard**.

Reviewer 4: ```suggestion
- **Create dashboards**. To learn more about creating dashboards in OpenSearch Dashboards, see **Dashboard**.
```

Reviewer 4: Should there be a link here?

Author: Added link.

---
layout: default

> title: Quickstart for OpenSearch Dashboards

Author: Change made.

nav_order: 20
has_children: false
---

# Quickstart for OpenSearch Dashboards

> This quickstart covers the core concepts for getting started with OpenSearch Dashboards. You'll learn how to:

Author: Change made

- Add sample data
- Explore and inspect data with **Discover**
- Visualize data with **Dashboard**

Before you get started, make sure you've installed OpenSearch and OpenSearch Dashboards. For information on installation and configuration, see [Install and configure OpenSearch]({{site.url}}{{site.baseurl}}/install-and-configure/install-opensearch/index/) and [Install and configure OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/install-and-configure/install-dashboards/index/).
{: .note}

# Adding sample data

Sample data sets come with visualizations, dashboards, and other tools to help you explore Dashboards before you add your own data. To add sample data, perform the following steps:

1. Verify access to OpenSearch Dashboards by connecting to [http://localhost:5601](http://localhost:5601) from a browser. The default username and password are `admin`.

> 1. On the OpenSearch Dashboards **Home** page, select **Add sample data**.

Author: Revised globally.

1. Select **Add data** to add the data sets.

>     <img src="{{site.url}}{{site.baseurl}}/images/add-sample-data.png" alt="Sample data sets" width="800">

Author: Revised

# Exploring and inspecting data

In [**Discover**](discover/index.md), you can: 

- Select data to explore, set a time range for that data, search it using [Dashboards Query Language (DQL)]({{site.url}}{{site.baseurl}}/dashboards/dql/), and filter the results.

> - Explore the data's details, view individual documents, and create tables summarizing the data's contents.

Author: Change made

- Visualize your findings.

## Try it: Getting familiar with Discover

1. On the OpenSearch Dashboards **Home** page, select **Discover.**
1. Change the [time filter](time-filter.md) to **Last 7 days**.

>     <img src="{{site.url}}{{site.baseurl}}/images/last-7--days.png" alt="Time filter interface" width="400"/>

Author: Global chage.

1. To view US-bound flights delayed 60 minutes or longer in a specific region, use the DQL search field:

    `FlightDelay:true AND DestCountry: US AND FlightDelayMin >= 60`.

> 1. To filter data by delay type, select the field under **Available fields**.

Author: The user can select any field they want. Revised for clarity.

>     <img src="{{site.url}}{{site.baseurl}}/images/filter-data.png" alt="Filter filter by Available fields" width="300"/>

Author: Global change made.

# Visualizing data

> Raw data can be hard to comprehend and use. Data visualizations help you prepare and present data in a visual form. In **Dashboard** you can:

Author: Change made

- Display data in a single view.
- Build dynamic dashboards.
- Create and share reports.
- Embed analytics to differentiate your applications.

## Try it: Getting familiar with Dashboard

1. On the OpenSearch Dashboards **Home** page, select **Dashboard**.
1. Select **[Flights] Global Flight Data** in the **Dashboards** window.

>     <img src="{{site.url}}{{site.baseurl}}/images/dashboard-flight-quickstart.png" alt="Data visualization dashboard" width="800"/>

Author: Global change made

1. To add panels to the dashboard, select **Edit** and then **Add** from the toolbar.

> 1. In the **Add panels** window, select the existing panel **[Flights] Delay Buckets**. You'll see a pop-up window in the bottom right confirming you've added the panel.

Author: Change made

1. Select `x` to close the Add panels window.
1. View the added panel **[Flights] Delay Buckets**, which is added as the last panel on the dashboard.

>     <img src="{{site.url}}{{site.baseurl}}/images/add-panel.png" alt="Add panel to dashboard" width="600"/>

Author: Global change made

## Try it: Creating a visualization panel

> Continuing with the preceding dashboard, you'll create a bar chart comparing the number of cancelled flights and delayed flights to delay type, and then add the panel to the dashboard.

Reviewer 4: ```suggestion
Continuing with the preceding dashboard, you'll create a bar chart comparing the number of canceled flights and delayed flights to delay type and then add the panel to the dashboard:
```

1. Change the default [time range]({{site.url}}{{site.baseurl}}/dashboards/get-started/time-filter/) from **24 hours** to **Last 7 days**. 
1. In the toolbar, select **Edit**, then **Create new**.
1. Select **VisBuilder** in the **New Visualizations** window.
1. In the **Data Source** dropdown list, select `opensearch_dashboards_sample_data_flights`.
1. Drag the fields **Cancelled** and **FlightDelay** to the y-axis column.
1. Drag the field **FlightDelayType** to the x-axis column.
1. Select **Save** and name the visualization in the **Title** field.

> 1. Select **Save and return**. The bar chart is added as the last panel to the dashboard.

Reviewer 4: ```suggestion
1. Select **Save and return**. The bar chart is added as the last panel on the dashboard.
```

> <img src="{{site.url}}{{site.baseurl}}/images/viz-panel-quickstart.png" alt="Creating a visualization panel" width="500"/>

Reviewer 4: Reference the image in the text that precedes it.

# Interacting with data

> Interactive dashboards allow you analyze data more deeply and filter it several ways. In Dashboards, you can interact directly with data on a dashboard by using dashboard-level filters. For example, continuing with the preceding dashboard, you can filter to show delays and cancellations for a specific airline.

Reviewer 4: ```suggestion
Interactive dashboards allow you analyze data in more depth and filter it in several ways. In Dashboards, you can interact directly with data on a dashboard by using dashboard-level filters. For example, continuing with the preceding dashboard, you can filter to show delays and cancellations for a specific airline.
```

## Try it: Interacting with the sample flight data

1. On the **[Flights] Airline Carrier** panel, select **OpenSearch-Air**. The dashboard updates automatically.
1. Select **Save** to save the customized dashboard.

Alternatively, you can apply filters using the dashboard toolbar:

1. In the dashboard toolbar, select **Add filter**.
1. From the **Field**, **Operator**, and **Value** dropdown lists, select **Carrier**, **is**, and **OpenSearch-Air**, respectively.

>     <img src="{{site.url}}{{site.baseurl}}/images/edit-filter.png" alt="Edit field interface" width="400"/>

Reviewer 4: Reference the image in the text that precedes it.

1. Select **Save**. The dashboard updates automatically.

> The result is the following dashboard: 

Reviewer 4: ```suggestion
The result is the dashboard shown in the following image.
```

  <img src="{{site.url}}{{site.baseurl}}/images/interact-filter-dashboard.png" alt="Dashboard view after applying Carrier filter" width="800"/>

# Taking the next steps

> **Visualize data**. Learn more about data visualizations in OpenSearch Dashboards. Go to [**Visualize**]({{site.url}}{{site.baseurl}}/dashboards/visualize/viz-index/).

Reviewer 4: ```suggestion
- To learn more about data visualizations in OpenSearch Dashboards, see [**Visualize**]({{site.url}}{{site.baseurl}}/dashboards/visualize/viz-index/).
```

> **Create dashboards.** Learn more about visualizing data in OpenSearch Dashboards. Go to [**Dashboard**](add link-content in development).

Reviewer 4: ```suggestion
- To learn more about creating dashboards in OpenSearch Dashboards, see [**Dashboard**](add link-content in development).
```

> **Explore data**. Learn more about exploring data in OpenSearch Dashboards. Go to [**Discover**]({{site.url}}{{site.baseurl}}/dashboards/discover/). 

Reviewer 4: ```suggestion
- To learn more about exploring data in OpenSearch Dashboards, see [**Discover**]({{site.url}}{{site.baseurl}}/dashboards/discover/). 
```

Reviewer 4: Changed to a simplified bulleted list. Not sure if you need to remove the empty lines.

Author: Change made. Removed empty lines
